sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidgetRenderer.kt

```diff
@@ -0,0 +1,307 @@
+package com.mapbox.maps.renderer.widget
+
+import android.graphics.Bitmap
+import android.opengl.GLES20
+import android.opengl.GLUtils
+import android.opengl.Matrix
+import com.mapbox.maps.renderer.gl.GlUtils
+import com.mapbox.maps.renderer.gl.GlUtils.toFloatBuffer
+import java.nio.FloatBuffer
+
+internal class BitmapWidgetRendererImpl(
+  private var bitmap: Bitmap,
+  private val position: WidgetPosition,
+  private val marginX: Float,
+  private val marginY: Float,
+) : WidgetRenderer {
+
+  private var program = 0
+  private var vertexShader = 0
+  private var fragmentShader = 0
+
+  private var attributeVertexPosition = 0
+  private var attributeTexturePosition = 0
+  private var uniformTexture = 0
+  private var uniformMvpMatrix = 0
+  private val textures = intArrayOf(0)
+
+  private var screenMatrix = FloatArray(0)
+  private val rotationMatrix = GlUtils.getIdentityMatrix()
+  private val translateMatrix = GlUtils.getIdentityMatrix()
+
+  private lateinit var mvpMatrixBuffer: FloatBuffer
+  private val mvpMatrix = GlUtils.getIdentityMatrix()
+
+  private var updateMatrix: Boolean = true
+
+  private lateinit var vertexPositionBuffer: FloatBuffer
+
+  private val texturePositionBuffer: FloatBuffer by lazy {
+    floatArrayOf(
+      0f, 0f,
+      0f, 1f,
+      1f, 0f,
+      1f, 1f
+    ).toFloatBuffer()
+  }
+
+  private var bitmapWidth = bitmap.width
+  private var bitmapHeight = bitmap.height
+
+  private var surfaceWidth = 0
+  private var surfaceHeight = 0
+
+  override var needRender: Boolean = true
+
+  override fun onSurfaceChanged(width: Int, height: Int) {
+    surfaceWidth = width
+    surfaceHeight = height
+
+    // transforms from (0,0) - (width, height) in screen pixels
+    // to (-1, -1) - (1, 1) for GL
+    screenMatrix = floatArrayOf(
+      2f / width, 0f, 0f, 0f,
+      0f, -2f / height, 0f, 0f,
+      0f, 0f, 0f, 0f,
+      -1f, 1f, 0f, 1f
+    )
+
+    Matrix.translateM(
+      translateMatrix,
+      0,
+      topX(),
+      leftY(),
+      0f
+    )
+
+    // in pixels, (-bitmapWidth / 2, -bitmapHeight/2) - (bitmapWidth / 2, bitmapHeight/2)
+    vertexPositionBuffer = floatArrayOf(
```

Comment:
Can we reuse same FloatBuffer and not create a new one?

Reply:
Reused buffer